tracker/graceful.go

```diff
@@ -0,0 +1,75 @@
+package tracker
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+
+	radio "github.com/R-a-dio/valkyrie"
+	"github.com/Wessie/fdstore"
+)
+
+type wireRecorder struct {
+	Listeners map[radio.ListenerClientID]*Listener
+}
+
+func (s *Server) storeSelf(ctx context.Context, fds *fdstore.Store) error {
+	err := fds.AddListener(s.httpLn, HttpLn, nil)
```

Comment:
use this data slot for the recorder state, you can marshal it to bytes right above it